duration.go

```diff
@@ -49,7 +49,7 @@ var (
 // Parse attempts to parse the given duration string into a *Duration,
 // if parsing fails an error is returned instead
 func Parse(d string) (*Duration, error) {
-	if !strings.Contains(d, "P") {
+	if !strings.HasPrefix(d, "P") && !strings.HasPrefix(d, "-P") {
```

Comment:
Use a switch it would be clearer

```go
switch {
   case 
         strings.HasPrefix(d, "P"),    // regular time period
         strings.HasPrefix(d, "-P"):  // negative time period

default:
        return nil, ErrUnexpectedInput
}

Reply:
This is a bit clearer 👍 Switched to this in the latest commit